src/file-events.c

```diff
@@ -413,6 +458,149 @@ int BPF_KPROBE(security_path_rename, const struct path *old_dir, struct dentry *
 }
 
 /* END RENAME PROBES */
+/* BEGIN OPEN-LIKE PROBES */
+
+SEC("tracepoint/sys_enter_open")
+int tracepoint__syscalls_sys_enter__open(struct syscalls_enter_open_args *ctx)
+{
+    if (is_write_open(ctx->flags)) {
+        enter_modify(ctx);
+        return 0;
+    }
+    u64 pid_tid = bpf_get_current_pid_tgid();
+    bpf_map_delete_elem(&incomplete_modifies, &pid_tid);
+    return 0;
+}
+
+SEC("tracepoint/sys_enter_openat")
+int tracepoint__syscalls_sys_enter__openat(struct syscalls_enter_openat_args *ctx)
+{
+    if (is_write_open(ctx->flags)) {
+        enter_modify(ctx);
+        return 0;
+    }
+    u64 pid_tid = bpf_get_current_pid_tgid();
+    bpf_map_delete_elem(&incomplete_modifies, &pid_tid);
+    return 0;
+}
+
+SEC("tracepoint/sys_enter_openat2")
+int tracepoint__syscalls_sys_enter__openat2(struct syscalls_enter_openat2_args *ctx)
+{
+    u64 flags = 0;
+    bpf_probe_read(&flags, sizeof(flags), &ctx->how->flags);
+    if (is_write_open(flags)) {
+        enter_modify(ctx);
+        return 0;
+    }
+    u64 pid_tid = bpf_get_current_pid_tgid();
+    bpf_map_delete_elem(&incomplete_modifies, &pid_tid);
+    return 0;
+}
+
+SEC("tracepoint/sys_enter_open_by_handle_at")
+int tracepoint__syscalls_sys_enter_open_by_handle_at(struct syscalls_enter_open_by_handle_at_args *ctx)
+{
+    if (is_write_open(ctx->flags)) {
+        enter_modify(ctx);
+        return 0;
+    }
+    u64 pid_tid = bpf_get_current_pid_tgid();
+    bpf_map_delete_elem(&incomplete_modifies, &pid_tid);
+    return 0;
+}
+
+SEC("kprobe/security_file_open")
+int BPF_KPROBE(security_file_open, void *file)
+{
+    void *path = ptr_to_field(file, CRC_FILE_F_PATH);
+    if (path == NULL)
+    {
+        file_message_t fm = {0};
+        push_file_warning(ctx, &fm, FM_MODIFY);
+        return 0;
+    }
+    store_modified_dentry(ctx, path);
+    return 0;
+}
+
+SEC("tracepoint/sys_exit_open")
+int tracepoint__syscalls_sys_exit__open(struct syscalls_exit_args *ctx)
+{
+    if (ctx->ret < 0)
+        return 0;
+    exit_modify(ctx);
+    return 0;
+}
+
+SEC("tracepoint/sys_exit_openat")
+int tracepoint__syscalls_sys_exit__openat(struct syscalls_exit_args *ctx)
+{
+    if (ctx->ret < 0)
+        return 0;
+    exit_modify(ctx);
+    return 0;
+}
+
+SEC("tracepoint/sys_exit_openat2")
+int tracepoint__syscalls_sys_exit__openat2(struct syscalls_exit_args *ctx)
+{
+    if (ctx->ret < 0)
+        return 0;
+    exit_modify(ctx);
+    return 0;
+}
+
+SEC("tracepoint/sys_exit_open_by_handle_at")
+int tracepoint__syscalls_sys_exit__open_by_handle_at(struct syscalls_exit_args *ctx)
+{
+    if (ctx->ret < 0)
+        return 0;
+    exit_modify(ctx);
+    return 0;
+}
+
+
+SEC("tracepoint/sys_enter_creat")
+int tracepoint__syscalls_sys_enter_creat(void *ctx)
+{
+    enter_create(ctx);
```

Comment:
creat doesn't always create, it's open and/or create so we need to look at the flags same we do for the other syscalls.

Reply:
I updated it to call `enter_modify`/`exit_modify`. `creat` doesn't get `flags` as an argument. The syscall just calls `sys_open`, `do_sys_open`, or `ksys_open` depending on kernel version with the flags set to (`O_CREAT | O_WRONLY | O_TRUNCATE`). Can see in [4.14](https://elixir.bootlin.com/linux/v4.14.32/source/fs/open.c#L1115:~:text=i386%20instead%3F%0A%20*/-,SYSCALL_DEFINE2,-(creat%2C) and [6.2](https://elixir.bootlin.com/linux/v6.2.8/source/fs/open.c#:~:text=i386%20instead%3F%0A%20*/-,SYSCALL_DEFINE2,-(creat%2C). So I believe we can just treat it with write flags always.